src/tools/concurrency/TraceParser.java

```diff
@@ -0,0 +1,342 @@
+package tools.concurrency;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+import som.VM;
+import som.vm.Symbols;
+import som.vm.VmSettings;
+import som.vmobjects.SSymbol;
+
+
+
+public final class TraceParser {
+  private static final byte ACTOR_CREATION = 1;
+  private static final byte PROMISE_CREATION = 2;
+  private static final byte PROMISE_RESOLUTION = 3;
+  private static final byte PROMISE_CHAINED = 4;
+  private static final byte MAILBOX = 5;
+  private static final byte THREAD = 6;
+  private static final byte MAILBOX_CONTD = 7;
+
+  private static final byte MESSAGE_BIT = (byte) 0x80;
```

Comment:
Why are these constants duplicated here? Can't we just put such things into a global place like `TraceData`? I suppose those constants need to remain in-sync across the whole program.